python/modules/IcePy/ObjectAdapter.cpp

```diff
@@ -29,18 +31,18 @@ static unsigned long _mainThreadId;
 namespace IcePy
 {
 
-typedef InvokeThread<Ice::ObjectAdapter> AdapterInvokeThread;
-typedef IceUtil::Handle<AdapterInvokeThread> AdapterInvokeThreadPtr;
-
 struct ObjectAdapterObject
 {
     PyObject_HEAD
     Ice::ObjectAdapterPtr* adapter;
-    IceUtil::Monitor<IceUtil::Mutex>* deactivateMonitor;
-    AdapterInvokeThreadPtr* deactivateThread;
+
+    std::future<void>* deactivateFuture;
+    std::exception_ptr* deactivateException;
```

Comment:
I find this pointer to a _ptr odd. Why not a `std::exception*`?

Reply:
It could just be a plain `Ice::Exception*`

Reply:
Changed to `Ice::Exception*`